articles/Monerium/monerium.md

```diff
@@ -0,0 +1,155 @@
+
+* [Another bank shall act as the custodian of the funds than the one used by the EMI.](https://eur-lex.europa.eu/legal-content/EN/TXT/?uri=CELEX%3A52020PC0593#:~:text=Issuers%20of%20significant%20asset%2Dreferenced%20tokens%20shall%20ensure,on%20a%20fair%2C%20reasonable%20and%20non%2Ddiscriminatory%20basis.)
+* [The equirement to monitor liquidity needs to meet ad-hoc redemption requests.](https://eur-lex.europa.eu/legal-content/EN/TXT/?uri=CELEX%3A52020PC0593#:~:text=Issuers%20of%20significant%20asset%2Dreferenced%20tokens%20shall%20assess%20and%20monitor%20the%20liquidity%20needs%20to%20meet%20redemption%20requests%20or%20the%20exercise%20of%20rights%2C%20as%20referred%20to%20in%20Article%2034%2C%20by%20holders%20of%20asset%2Dreferenced%20tokens.)
+* [The over-collateralization percentage shall be 3% instead of 2%](https://eur-lex.europa.eu/legal-content/EN/TXT/?uri=CELEX%3A52020PC0593#:~:text=The%20percentage%20referred%20to%20in%20Article%C2%A031(1)%2C%20point%20(b)%2C%20shall%20be%20set%20at%203%25%20of%20the%20average%20amount%20of%20the%20reserve%20assets%20for%20issuers%20of%20significant%20asset%2Dreferenced%20tokens.)
+* [A plan should be in place to be able to orderly wind down the company's activities.](https://eur-lex.europa.eu/legal-content/EN/TXT/?uri=CELEX%3A52020PC0593#:~:text=Chapter%206-,Orderly%20wind%2Ddown,-Article%2042%0AOrderly)
+
+### Smart Contracts
+
+There were no smart contract audits being provided for Monerium's EURe. However, we decided to analyze the contracts ourselves, specifically looking for centralization vectors and elevated rights.
+
+What we found was the following:
+
+* [The owner](https://etherscan.io/address/0xA176a20E3698bAa36Ea09E4499e407e7dbFdfa0c) of the [EURe contract](https://etherscan.io/address/0x3231cb76718cdef2155fc47b5286d82e6eda273f#writeContract) is an externally owned account (EOA)
+* [The owner](https://etherscan.io/address/0xA176a20E3698bAa36Ea09E4499e407e7dbFdfa0c) of the [EURe contract](https://etherscan.io/address/0x3231cb76718cdef2155fc47b5286d82e6eda273f#writeContract) has the ability to upgrade the controller. (And hence being able to infinitely mint EURe)
+* [The owner](https://etherscan.io/address/0xA176a20E3698bAa36Ea09E4499e407e7dbFdfa0c) of the [EURe contract](https://etherscan.io/address/0x3231cb76718cdef2155fc47b5286d82e6eda273f#writeContract) has the ability to call the `selfdestruct` function, which kills the contract and transfers all EURe to the owner.
+* [A SystemAccount](https://etherscan.io/address/0x882145b1f9764372125861727d7be616c84010ef) (within the controller) is an EOA.
+* [A SystemAccount](https://etherscan.io/address/0x882145b1f9764372125861727d7be616c84010ef) is able to (infinitely) mint EURe.
+* [The controller contract](https://etherscan.io/address/0x914C6a264bC3cC68966C33BBD6630F011E7a47cF) is not verified on etherscan.
```

Comment:
Should make sure they verify all contracts on Etherscan.  This is a red flag.

Reply:
Yes, we will. 

At the time of deployment, we only verified the EURe token itself. We will verify the rest of the contracts which comprise the token system. Unfortunately, this is not as straightforward as it should be, because of different compiler versions, maximum library dependencies (10) by the Etherscan verifier, etc. We will however find a solution to verifying all the accounts.